operators/src/engine/operator.rs

```diff
@@ -162,6 +165,31 @@ pub trait InitializedVectorOperator: Send + Sync {
     {
         Box::new(self)
     }
+
+    /// Get a canonic representation of the operator and its sources.
+    /// This only includes *logical* operators, not wrappers
+    fn canonic_name(&self) -> CanonicOperatorName;
+}
+
+/// A canonic name for an operator and its sources
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct CanonicOperatorName(serde_json::Value);
```

Comment:
Maybe we could try to use a more compact representation as text?

https://serde.rs/#data-formats

Reply:
I am not sure which one to use. From the native supported CBOR and MessagePack could be suited.

Also https://crates.io/crates/bincode sounds promising.

I think one benefit of sticking with Json Value is that it's maybe faster to compare two values. But I could be wrong.

Should I just pick any one or should I compare them or do we make a new task?

Reply:
Shouldn't `Eq` be faster the less bytes you have?

You could try it out and just pick one.

Msg. me for merging this stuff.

Reply:
If we stick with JSON, should we put the serialized bytes in here?

For computing the hash, Value::Map would go over this:

https://doc.rust-lang.org/src/alloc/collections/btree/map.rs.html#2151

Or is it cheaper to not make a string out of it?

Is it cheaper to serialize to Value than to string?

Reply:
> If we stick with JSON, should we put the serialized bytes in here?

I still think it's handy if the name is human readable without having to deserialize it

> For computing the hash, Value::Map would go over this:

what do you mean? A serde json Map is not a BtreeMap

> Or is it cheaper to not make a string out of it?

we need to have something that can be hashed. A String can be hasehd, a json value cannot.

> Is it cheaper to serialize to Value than to Vec?

I don't know, shall I benchmark it?

Reply:
> > If we stick with JSON, should we put the serialized bytes in here?
> 
> I still think it's handy if the name is human readable without having to deserialize it
if you serialize it to bytes, it is still human readable

> 
> > For computing the hash, Value::Map would go over this:
> 
> what do you mean? A serde json Map is not a BtreeMap
Well… it 's just that it is.

> 
> > Or is it cheaper to not make a string out of it?
> 
> we need to have something that can be hashed. A String can be hasehd, a json value cannot.
Shouldn't we just store a hashable and comparable thing like a string or a Vec<u8> instead of a value that we always `to_string` for each hash-call?

> 
> > Is it cheaper to serialize to Value than to Vec?
> 
> I don't know, shall I benchmark it?
We could look what the code does.